pkgdb/src/util.cc

```diff
@@ -70,6 +71,21 @@ isSQLiteDb( const std::string & dbPath )
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+bool
+isSQLError( int rcode )
```

Comment:
Would an `isSQLSuccess` be more useful and have it return failure for BUSY as well.  In the even you need to distinguish between BUSY and ERROR, then you can check, but is seem like in most places a single SUCCESS check would do the trick and might be more concise and readable.

Reply:
maybe, I realized it already treated `SQLITE_BUSY` as a failure so I was able to drop the explicit `SQLITE_BUSY` checks I had added.

Does that resolve the concerns about it being concise?

If not we can definitely replace those with a `isSQLSuccess()` function instead.